src/render_helpers/surface.rs

```diff
@@ -53,17 +55,37 @@ pub fn render_snapshot_from_surface_tree(
                 }
 
                 let data = data.lock().unwrap();
-                let Some(texture) = data.texture(renderer.context_id()) else {
-                    return;
-                };
+                let buffer = {
+                    if let Some(texture) = data.texture::<GlesTexture>(renderer.context_id()) {
+                        TextureBuffer::from_texture(
+                            renderer,
+                            texture.clone(),
+                            f64::from(data.buffer_scale()),
+                            data.buffer_transform(),
+                            Vec::new(),
+                        )
+                    } else if let Some(single_pixel_buffer_user_data) = data
+                        .buffer()
+                        .and_then(|buffer| get_single_pixel_buffer(buffer).ok())
+                    {
+                        let pixel: [u8; 4] = single_pixel_buffer_user_data.rgba8888();
+                        let argb_pixel = [pixel[3], pixel[1], pixel[2], pixel[0]];
```

Comment:
Is this premultiplied? Does TextureBuffer::from_memory expect premultiplied? Add a comment here pls

Reply:
my cursory look through `gles/mod.rs` shows that the renderer implementation expects premultiplied because it uses ` self.gl.BlendFunc(ffi::ONE, ffi::ONE_MINUS_SRC_ALPHA);` and `TextureBuffer` when used for `TextureRenderElement` passes the imported texture to the renderer with no real modification after import.